node/src/actors/chain_manager/mod.rs

```diff
@@ -1736,6 +1754,97 @@ impl ChainManager {
             Ok(hashes)
         }
     }
+
+    /// This function takes the transactions that were included in the mempool during an unconfirmed
+    /// superepoch and the transactions included in unconfirmed blocks and it allows them to be
+    /// included again in the mempool when the node wold be Synced again
+    #[must_use]
+    fn reinsert_unconfirmed_transactions(
+        &mut self,
+        epoch: Epoch,
+    ) -> ResponseActFuture<Self, (), ()> {
+        let inventory_manager = InventoryManager::from_registry();
+
+        let mut mempool_transactions = self.transactions_pool.remove_unconfirmed_transactions();
```

Comment:
We need to double check that it's impossible to add a transaction to the `TransactionsPool` between this call, which removes the unconfirmed transactions, and the call to `initialize_from_storage()` which restores the previous `ChainState`. Otherwise it could be possible to insert a transaction which is not valid after the revert.